### ISSUE TITLE
use of compare_exchange_weak

### ISSUE THREAD

**Issue by Philippe91** [rigtorp/MPMCQueue#47]
Both for enqueing and dequeing, why not using `compare_exchange_weak / std::memory_order_relaxed` (instead of `compare_exchange_strong / memory_order_seq_cst`) like in Dmitry Vyukov latest implementation (2021)?
This seems to make sense, as we are in a loop.

https://drive.google.com/file/d/1uCefvM3bTnWLFrcYoMxCOKGjWwHJQM2n/view

**Comment by rigtorp** [comment]
yes I haven't optimized for ARM. Ideally there should only be single dmb
barrier.
